File: Statista.Domain/Entities/Question/Section.cs
namespace Statista.Domain.Entities;

public class Section
{
    public Guid Id { get; set; }
    public string Title { get; set; }
    public ICollection<Section> sections { get; set; }
    public ICollection<Question> questions { get; set; }
    public int? order { get; set; }
}

File: statistika_mobile/lib/features/form/domain/model/section.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:statistika_mobile/features/form/domain/model/question.dart';

part 'section.g.dart';

@JsonSerializable()
class Section {
  Section({
    required this.id,
    required this.title,
    required this.formId,
    required this.order,
    required this.questions,
  });

  final String id;
  final String? title;
  final String formId;
  final int? order;
  final List<Question> questions;

  factory Section.fromJson(Map<String, dynamic> json) =>
      _$SectionFromJson(json);

  Map<String, dynamic> toJson() => _$SectionToJson(this);
}
